### PR TITLE
feat(connector): [Ebanx] Add payout flows 

### PR DESCRIPTION
**Pull request by Sakilmostak**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Payout create, Payout commit and Payout Cancel flows are added to Ebanx

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

It is a closed sandbox thus no method is available for testing

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by pixincreate** [comment]
Please mark this PR `draft` if still under development.

File: crates/api_models/src/payouts.rs [hunk 1]
```diff
@@ -261,6 +262,28 @@ pub struct SepaBankTransfer {
...
+    /// Bank account number is an unique identifier assigned by a bank to a customer.
+    #[schema(value_type = String, example = "000123456")]
+    pub bank_account_number: Secret<String>,
+
+    /// Unique key for pix customer
+    #[schema(value_type = String, example = "000123456")]
+    pub pix_key: Secret<String>,
+
+    /// Individual taxpayer identification number
+    pub tax_id: Secret<String>,
```
Comment (SamraatBansal) [inline crates/api_models/src/payouts.rs:284]: Will this always exist regardless of the country? Or this is LATAM Specific, please check if this is required to be made optional

File: crates/router/src/connector/ebanx.rs [hunk 2]
```diff
@@ -111,113 +104,83 @@ impl ConnectorCommon for Ebanx {
...
-
-impl
-    ConnectorIntegration<
-        api::SetupMandate,
-        types::SetupMandateRequestData,
-        types::PaymentsResponseData,
-    > for Ebanx
-{
-}
+#[async_trait::async_trait]
```
Comment (SamraatBansal) [inline crates/router/src/connector/ebanx.rs:118]: Async trait is not required, there is no async function inside it

File: crates/router/src/connector/ebanx.rs [hunk 3]
```diff
@@ -227,142 +190,83 @@ impl ConnectorIntegration<api::Authorize, types::PaymentsAuthorizeData, types::P
...
+            )?)
+            .set_body(types::PayoutFulfillType::get_request_body(
+                self, req, connectors,
+            )?)
+            .build();
+
+        Ok(Some(request))
     }
 
+    #[instrument(skip_all)]
```
Comment (SamraatBansal) [inline crates/router/src/connector/ebanx.rs:255]: Why the tracing is skipped?

**Review by SamraatBansal — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/ebanx/transformers.rs [hunk 4]
```diff
@@ -1,237 +1,414 @@
...
+                    bank_name: pix_data.bank_name,
+                    account_type: Some(EbanxBankAccountType::CheckingAccount),
+                };
+
+                let billing_address = item.router_data.get_billing_address()?;
+                let customer_details = item.router_data.request.customer_details.clone().ok_or(
+                    errors::ConnectorError::MissingRequiredField {
+                        field_name: "Customer Details",
+                    },
+                )?;
```
Comment (srujanchikke) [inline crates/router/src/connector/ebanx/transformers.rs:119]: Can we move this to utils ?

File: crates/router/src/connector/ebanx/transformers.rs [hunk 5]
```diff
@@ -1,237 +1,414 @@
...
                 Ok(Self {
-                    amount: item.amount.to_owned(),
-                    card,
+                    amount: item.amount,
+                    integration_key: ebanx_auth_type.integration_key,
+                    country: customer_details.phone_country_code.ok_or(
+                        errors::ConnectorError::MissingRequiredField {
+                            field_name: "Customer Details",
+                        },
+                    )?,
```
Comment (srujanchikke) [inline crates/router/src/connector/ebanx/transformers.rs:143]: Can we move this to utils ?

File: crates/router/src/connector/ebanx/transformers.rs [hunk 6]
```diff
@@ -1,237 +1,414 @@
...
+                    target: EbanxPayoutType::PixKey,
+                    target_account: pix_data.pix_key,
+                    payee,
                 })
             }
-            _ => Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into()),
+            _ => Err(errors::ConnectorError::NotSupported {
+                message: "Payment Method Not Supported".to_string(),
+                connector: "Ebanx",
+            })?,
```
Comment (srujanchikke) [inline crates/router/src/connector/ebanx/transformers.rs:154]: Can we not use defaults

File: crates/router/src/connector/ebanx/transformers.rs [hunk 7]
```diff
@@ -1,237 +1,414 @@
...
-                connector_refund_id: item.response.id.to_string(),
-                refund_status: enums::RefundStatus::from(item.response.status),
+            response: Ok(types::PayoutsResponseData {
+                status: Some(storage_enums::PayoutStatus::from(item.response.status)),
+                connector_payout_id: item
+                    .data
+                    .request
+                    .connector_payout_id
+                    .clone()
+                    .ok_or(errors::ConnectorError::MissingConnectorTransactionID)?,
```
Comment (srujanchikke) [inline crates/router/src/connector/ebanx/transformers.rs:314]: Can we move this to utils ?

**Review by srujanchikke — CHANGES_REQUESTED** [review]

**Review by srujanchikke — DISMISSED** [review]

File: crates/router/src/connector/ebanx/transformers.rs [hunk 8]
```diff
@@ -1,237 +1,404 @@
...
         }
     }
 }
 
-//TODO: Fill the struct with respective fields
-// Auth Struct
 pub struct EbanxAuthType {
-    pub(super) api_key: Secret<String>,
+    #[allow(dead_code)]
+    pub(super) integration_key: Secret<String>,
```
Comment (SamraatBansal) [inline crates/router/src/connector/ebanx/transformers.rs:156]: Why dead code?

**Review by SamraatBansal — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/ebanx/transformers.rs [hunk 9]
```diff
@@ -1,237 +1,406 @@
...
+                    bank_branch: pix_data.bank_branch,
+                    bank_name: pix_data.bank_name,
+                    account_type: Some(EbanxBankAccountType::CheckingAccount),
+                };
+
+                let billing_address = item.router_data.get_billing_address()?;
+                let customer_details = item.router_data.request.get_customer_details()?;
+
+                let document_type = pix_data.tax_id.clone().map(|tax_id| {
+                    if tax_id.clone().expose().len() == 11 {
```
Comment (SamraatBansal) [inline crates/router/src/connector/ebanx/transformers.rs:119]: Is this the correct and robust way to deal?

**Review by SamraatBansal — DISMISSED** [review]

File: crates/router/src/core/payment_methods/vault.rs [hunk 10]
```diff
@@ -483,6 +483,8 @@ pub struct TokenizedBankSensitiveValues {
     pub bic: Option<masking::Secret<String>>,
     pub bank_sort_code: Option<masking::Secret<String>>,
     pub iban: Option<masking::Secret<String>>,
+    pub pix_key: Option<masking::Secret<String>>,
```
Comment (kashif-m) [inline crates/router/src/core/payment_methods/vault.rs:486]: Let's refactor this later.

**Review by kashif-m — APPROVED** [review]

**Review by srujanchikke — DISMISSED** [review]

File: crates/router/src/connector/ebanx/transformers.rs (hunk 9 above)
Comment (Sakilmostak) [reply crates/router/src/connector/ebanx/transformers.rs:119]: the tax id comes in a specific format of fixed length, thus, this should be able to handle every cases, you can refer to this article for more [info](https://wise.com/gb/blog/cpf-cnpj-meaning-brazil#:~:text=What%20is%20the%20difference%20between,at%20the%20time%20of%20formation.)

File: crates/api_models/src/payouts.rs [hunk 11]
```diff
@@ -261,6 +262,29 @@ pub struct SepaBankTransfer {
...
+    /// Bank account number is an unique identifier assigned by a bank to a customer.
+    #[schema(value_type = String, example = "000123456")]
+    pub bank_account_number: Secret<String>,
+
+    /// Unique key for pix customer
+    #[schema(value_type = String, example = "000123456")]
+    pub pix_key: Secret<String>,
+
+    /// Individual taxpayer identification number
+    #[schema(value_type = String, example = "000123456")]
```
Comment (Narayanbhat166) [inline crates/api_models/src/payouts.rs:284]:
```suggestion
    #[schema(value_type = Option<String>, example = "000123456")]
```

**Review by Narayanbhat166 — APPROVED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

**Review by srujanchikke — APPROVED** [review]

**Review by kashif-m — APPROVED** [review]

**Pull request merged**